python/pyarrow/tests/test_compute.py

```diff
@@ -2217,3 +2217,12 @@ def test_list_element():
     result = pa.compute.list_element(lists, index)
     expected = pa.array([{'a': 5.6, 'b': 6}, {'a': .6, 'b': 8}], element_type)
     assert result.equals(expected)
+
+
+def test_count_distinct():
+    seed = datetime.now()
+    samples = [seed.replace(year=y) for y in range(1992, 2092)]
+    arr = pa.array(samples, pa.timestamp("ns"))
+    result = pa.compute.count_distinct(arr)
+    expected = pa.scalar(len(samples), type=pa.int64())
```

Comment:
Add tests using the `CountOptions`, see [`count`](https://github.com/apache/arrow/blob/master/python/pyarrow/tests/test_compute.py#L1839).

Reply:
Tests for count options are in the c++ side.

Reply:
In general, Python tests should make use of the features included in the Python bindings, if there is none then there is no need for a test case because it falls back to C++ tests. In this case, there is a Python binding for `CountOptions` so you should test based on this.